software/firmware/bootloader.py

```diff
@@ -144,4 +159,11 @@ def main(self):
             europi.b1._handler_both(europi.b2, self.exit_to_menu)
             europi.b2._handler_both(europi.b1, self.exit_to_menu)
 
-            script_class().main()
+            try:
+                script_class().main()
+            except Exception:
+                # in case we have the USB cable connected, print the stack trace for debugging
+                # otherwise, just halt and show the error message
+                print(f"[ERR ] Failed to run script: {err}")
+                traceback.print_exc()
+                self.show_error("Crash", f"Script died\n{type(err)}", -1)
```

Comment:
[optional] The phrases 'Crash' and 'Script died' are redundant and take up valuable screen real estate. I would suggest removing at least one of those lines and including more of the error message itself.

Reply:
Refactored this to show as much of the error message as fits on the screen. I'd originally tried to keep it short to avoid truncating text, but you're probably right that some detail is better than a generic "something died" message.